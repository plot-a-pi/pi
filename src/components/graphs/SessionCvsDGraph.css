.SessionCvsDGraph {
  margin: auto;
  margin-bottom: 30vw;

  @media screen and (min-aspect-ratio: 3/2) {
    width: 75%;
  }
}