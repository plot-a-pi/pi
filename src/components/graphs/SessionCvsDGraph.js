import React, { useEffect } from 'react';
import CvsDScatterplot from './CvsDScatterplot';
import PropTypes from 'prop-types';
import { useEmitEvent, useSocket, useSocketState } from 'react-socket-io-hooks';
import styles from './SessionCvsDGraph.css';

const SessionCvsDGraph = ({ match }) => {
  const { id } = match.params;
  const socket = useSocket();

  const emitJoinSession = useEmitEvent('JOIN_SESSION');
  const emitSessionData = useEmitEvent('RETRIEVE_SESSION_DATA');
  const emitSessionStats = useEmitEvent('GET_SESSION_STATS');
  const { sessionData, sessionStats } = useSocketState();

  useEffect(() => {

    if(socket.connected !== undefined) {
      emitJoinSession(id);
      emitSessionData(id);
      emitSessionStats(id);
    }

  }, [socket.connected]);

  if(!sessionData || !sessionData[0] || !sessionStats || !sessionStats.mean) return null;

  return (
    <div className={styles.SessionCvsDGraph}>
      <CvsDScatterplot data={sessionData[0]} stats={sessionStats} title={'Class Circle Measurement Data'} xLabel={'Diameter (in)'} yLabel={'Circumference (in)'} line={false} />
    </div>
  );

};
SessionCvsDGraph.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    })
  }).isRequired
};

export default SessionCvsDGraph;
