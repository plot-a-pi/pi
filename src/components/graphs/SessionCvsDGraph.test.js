import React from 'react';
import { shallow } from 'enzyme';
import SessionCvsDGraph from './SessionCvsDGraph';

jest.mock('react-socket-io-hooks', () => ({
  useEmitEvent: () => {}, 
  useSocketState: () => ({
    recentUserPoint: {}
  }), 
  useSocket: () => ({
    socket: {
      connected: null
    }
  }),
}));

describe('SessionCvsDGraph component', () => {
  it('renders SessionCvsDGraph', () => {
    const wrapper = shallow(<SessionCvsDGraph match={{ params: { id: '1' } }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
