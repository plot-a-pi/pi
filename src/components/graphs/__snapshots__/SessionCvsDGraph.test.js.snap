// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvsDScatterplot component renders CvsDScatterplot 1`] = `
<div
  className="container"
>
  <svg
    className="svg"
  >
    <text
      className="title"
      fill="#212E59"
    >
      test title
    </text>
    <g
      className="x-axis"
    />
    <text
      className="x-label"
      fill="#212E59"
    >
      test x-label
    </text>
    <g
      className="y-axis"
    />
    <text
      className="y-label"
      fill="#212E59"
    >
      test y-label
    </text>
    <line
      className="line"
    />
  </svg>
  <div
    className="csvButton"
  >
    <CSVButton
      data={
        Array [
          Array [
            undefined,
            undefined,
          ],
        ]
      }
      header1="Diameter (in)"
      header2="Circumference (in)"
    />
  </div>
</div>
`;

exports[`SessionCvsDGraph component renders SessionCvsDGraph 1`] = `""`;
